common/src/main/java/com/wynntils/mc/render/RenderUtils.java

```diff
@@ -383,6 +406,68 @@ public static void drawArc(
         RenderSystem.disableBlend();
     }
 
+    public static void drawRoundedRectWithBorder(
+            PoseStack poseStack,
+            CustomColor borderColor,
+            CustomColor fillColor,
+            float x,
+            float y,
+            float z,
+            float width,
+            float height,
+            float lineWidth,
+            int innerRadius,
+            int outerRadius) {
+
+        float x2 = x + width;
+        float y2 = y + height;
+
+        // Fill the rect
+        final int fillOffset = (int) lineWidth;
+        drawRect(
+                poseStack,
+                fillColor,
+                x + fillOffset,
+                y + fillOffset,
+                z,
+                width - fillOffset * 2,
+                height - fillOffset * 2);
+        drawLine(poseStack, fillColor, x + fillOffset, y + fillOffset, x2 - fillOffset, y + fillOffset, z, lineWidth);
+        drawLine(poseStack, fillColor, x2 - fillOffset, y + fillOffset, x2 - fillOffset, y2 - fillOffset, z, lineWidth);
+        drawLine(poseStack, fillColor, x + fillOffset, y2 - fillOffset, x2 - fillOffset, y2 - fillOffset, z, lineWidth);
+        drawLine(poseStack, fillColor, x + fillOffset, y + fillOffset, x + fillOffset, y2 - fillOffset, z, lineWidth);
+
+        float offset = outerRadius - 1;
+
+        // Edges
+        drawLine(poseStack, borderColor, x + offset, y, x2 - offset, y, z, lineWidth);
+        drawLine(poseStack, borderColor, x2, y + offset, x2, y2 - offset, z, lineWidth);
+        drawLine(poseStack, borderColor, x + offset, y2, x2 - offset, y2, z, lineWidth);
+        drawLine(poseStack, borderColor, x, y + offset, x, y2 - offset, z, lineWidth);
+
+        // Corners
+        poseStack.pushPose();
+        poseStack.translate(-1, -1, 0);
+        drawRoundedCorner(poseStack, borderColor, x, y, z, innerRadius, outerRadius, Mth.HALF_PI * 3);
```

Comment:
`Mth`?

Reply:
Mth is Minecraft's MathUtils class.

Reply:
Name of minecraft math class.

Reply:
Ah. Weird name. It used to be MathUtils, iirc. Much better imho.